### PR TITLE
Update getThreadState to handle unmounted carrier thread

### PR DESCRIPTION
**Pull request by babsingh**
If a virtual thread is mounted on a carrier thread, then the carrier
thread is considered implicitly parked.

We match the RI behaviour and pass `vthread/ThreadStateTest` with
these changes.

Related: #18054

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
@gacholio Requesting your review.

File: runtime/jvmti/jvmtiHelpers.c [hunk 1]
```diff
@@ -700,10 +700,9 @@ javaOffloadSwitchOn(J9VMThread * currentThread, UDATA eventNumber, UDATA javaOff
 
 
 jint
-getThreadState(J9VMThread *currentThread, j9object_t threadObject)
+getThreadStateHelper(J9VMThread *currentThread, j9object_t threadObject, J9VMThread *vmThread)
```
Comment (gacholio) [inline runtime/jvmti/jvmtiHelpers.c:703]: Why break this into it's own function rather than just adding the new code at the end?

Comment (babsingh) [reply runtime/jvmti/jvmtiHelpers.c:703]: Now, `getThreadState` is strictly used with platform (non-virtual) threads. `getVirtualThreadState` used to rely on `getThreadState`, which has been changed to the helper. `getVirtualThreadState` malfunctions if the new code, which is platform thread specific, is added at the end.

[the remark this answers is not in the document]
Comment (babsingh) [reply runtime/jvmti/jvmtiHelpers.c:703]: Yes, I have changed it to a static function.

**Review by gacholio — APPROVED** [review]

**Pull request merged**